includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -643,16 +643,49 @@ protected function update_address( $order, $posted, $type = 'billing' ) {
 	 * @return int
 	 */
 	protected function get_product_id( $posted, $action = 'create' ) {
-		if ( ! empty( $posted['sku'] ) ) {
-			$product_id = (int) wc_get_product_id_by_sku( $posted['sku'] );
-		} elseif ( ! empty( $posted['product_id'] ) && empty( $posted['variation_id'] ) ) {
-			$product_id = (int) $posted['product_id'];
-		} elseif ( ! empty( $posted['variation_id'] ) ) {
-			$product_id = (int) $posted['variation_id'];
-		} elseif ( 'update' === $action ) {
-			$product_id = 0;
+
+		$has_filter_applied = has_filter( 'wc_product_has_unique_sku', '__return_false' );
```

Comment:
I believe this only works if people used the shorthand boolean functions. If they wrote their own custom function, this would return false. Perhaps skip the second argument and check for `true` and `> 0`